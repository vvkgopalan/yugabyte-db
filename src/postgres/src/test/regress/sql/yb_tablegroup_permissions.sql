--
-- YB_TABLEGROUP Testsuite: Testing permissions for TABLEGROUP.
--
CREATE DATABASE test_tablegroup;
\c test_tablegroup
--
-- CREATE TABLEGROUP
--
CREATE USER user_1;
CREATE USER user_2 SUPERUSER;
CREATE USER user_3;
GRANT CREATE ON DATABASE test_tablegroup TO user_3;
CREATE TABLEGROUP owned_by1 OWNER user_1;
CREATE TABLEGROUP supergroup;
--
-- Operations with user_3
--
\c test_tablegroup user_3
-- Create tablegroup should succeed as well as dropping that tablegroup.
CREATE TABLEGROUP owned_by3;
DROP TABLEGROUP owned_by3;
-- However these should fail.
DROP TABLEGROUP supergroup;
DROP TABLEGROUP owned_by1;
--
-- Operations with user_2
--
\c test_tablegroup user_2
-- All operations should succeed.
CREATE TABLEGROUP owned_by2;
CREATE TABLEGROUP owned_by1_created_by2 OWNER user_1;
DROP TABLEGROUP supergroup;
CREATE TABLE a(i text) TABLEGROUP owned_by2;
--
-- Operations with user_1
--
\c test_tablegroup user_1
-- These will fail
DROP TABLEGROUP owned_by2;
CREATE TABLEGROUP owned_by1_created_by1;
-- However these should succed
DROP TABLEGROUP owned_by1;
DROP TABLEGROUP owned_by1_created_by2;
-- Create table with a tablegroup should default to no privs. Only owner/superuser.
-- This should fail.
CREATE TABLE b(i text) TABLEGROUP owned_by2;
