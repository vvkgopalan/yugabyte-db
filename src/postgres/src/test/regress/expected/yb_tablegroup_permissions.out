--
-- YB_TABLEGROUP Testsuite: Testing permissions for TABLEGROUP.
--
CREATE DATABASE test_tablegroup;
\c test_tablegroup
--
-- CREATE TABLEGROUP
--
CREATE USER user_1;
CREATE USER user_2 SUPERUSER;
CREATE USER user_3;
GRANT CREATE ON DATABASE test_tablegroup TO user_3;
CREATE TABLEGROUP owned_by1 OWNER user_1;
CREATE TABLEGROUP supergroup;
--
-- Operations with user_3
--
\c test_tablegroup user_3
-- Create tablegroup should succeed as well as dropping that tablegroup.
CREATE TABLEGROUP owned_by3;
DROP TABLEGROUP owned_by3;
CREATE TABLEGROUP owned_by1_created_by3 OWNER user_1;
-- However these should fail.
DROP TABLEGROUP supergroup;
ERROR:  must be owner of tablegroup supergroup
DROP TABLEGROUP owned_by1;
ERROR:  must be owner of tablegroup owned_by1
--
-- Operations with user_2
--
\c test_tablegroup user_2
-- All operations should succeed.
CREATE TABLEGROUP owned_by2;
CREATE TABLEGROUP owned_by1_created_by2 OWNER user_1;
DROP TABLEGROUP supergroup;
CREATE TABLE a(i text) TABLEGROUP owned_by2;
--
-- Operations with user_1
--
\c test_tablegroup user_1
-- These will fail
DROP TABLEGROUP owned_by2;
ERROR:  must be owner of tablegroup owned_by2
CREATE TABLEGROUP owned_by1_created_by1;
ERROR:  permission denied for database test_tablegroup
-- However these should succed
DROP TABLEGROUP owned_by1;
DROP TABLEGROUP owned_by1_created_by2;
CREATE TABLE b(i text) TABLEGROUP owned_by1_created_by3;
-- Create table/index with a tablegroup should default to no privs. Only owner/superuser.
-- This should fail.
CREATE TABLE c(i text) TABLEGROUP owned_by2;
ERROR:  permission denied for tablegroup owned_by2
CREATE INDEX ON b(i) TABLEGROUP owned_by2;
ERROR:  permission denied for tablegroup owned_by2
